src/base/static/utils/geospatial-utils.js

```diff
@@ -0,0 +1,183 @@
+            feature.properties[config.aggregator.property]
+          ]
+            ? ++memo[feature.properties[config.aggregator.property]]
+            : 1,
+        }),
+        {},
+      );
+
+      return {
+        [config.name]: uniqueValues,
+      };
+    } else {
+      return {
+        [config.name]: pointsWithin.features.length,
+      };
+    }
+  },
+  aggregatePolygonsOverlappingBuffer: ({
+    placeGeometry,
+    sourceFeatures,
+    config,
+  }) => {
+    const bufferFeature = getBufferFeature(placeGeometry, config.buffer);
+
+    if (!bufferFeature) {
+      return {};
+    }
+
+    sourceFeatures = sourceFeatures.reduce((memo, feature) => {
+      if (feature.geometry.type === "MultiPolygon") {
+        // Flatten any MultiPolygons to arrays of individual Polygon features,
+        // because `booleanOverlap` can only compare like feature types.
+        try {
+          feature = flatten(feature).features;
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
+
+          feature = [];
+        }
+      }
+
+      return memo.concat(feature);
+    }, []);
+
+    let overlappingFeatures = sourceFeatures.filter(feature => {
+      try {
+        return (
+          // Technically, we're doing more than an overlap comparison here.
+          // `boooleanOverlap` returns features which are strictly overlapping; we
+          // also want to return features which are entirely contained by the
+          // buffer.
+          booleanOverlap(bufferFeature, feature) ||
+          booleanContains(bufferFeature, feature)
+        );
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error(e);
+
+        return false;
+      }
+    });
+
+    const propertyHashes = new Set();
+    overlappingFeatures = overlappingFeatures.filter(overlappingFeature => {
+      // We need to de-dupe the array of `overlappingFeatures`. Duplicate
+      // features might exist because Mapbox's `querySourceFeatures()` method
+      // sometimes returns duplicates due to the nature of vector tiles.
+      // See: https://docs.mapbox.com/mapbox-gl-js/api/#map#querysourcefeatures
+      //
+      // Duplicates might also exist as a result of flattening a MultiPolygon
+      // above. We determine duplicate features by comparing hashes of their
+      // properties objects, since we don't have reliable access to a unique
+      // feature id for third-party data sources.
+      const propertyHash = hash(overlappingFeature.properties);
+
+      if (propertyHashes.has(propertyHash)) {
+        return false;
+      }
+
+      propertyHashes.add(propertyHash);
+
+      return true;
+    });
+
+    if (config.aggregator.type === "totalCount") {
+      return {
+        [config.name]: overlappingFeatures.length,
+      };
+    } else {
+      return {
+        [config.name]: overlappingFeatures.length,
+      };
+    }
```

Comment:
nit: I can see how you wanted to make this a safe fallback, but I think it would be better to throw an error in the `else` block, or not have it at all.

I don't think we need to change this right now. But we can enforce the invariant the `config.aggregator.type` must be equal to "totalCount" or "anotherString"  in TS using enums, or perhaps we can do it using prop types in our `placeConfigPropType`.